elasticapm/handlers/logging.py

```diff
@@ -155,3 +156,51 @@ def _emit(self, record, **kwargs):
             logger_name=record.name,
             **kwargs
         )
+
+
+class LoggingFilter(logging.Filter):
+    """
+    This filter doesn't actually do any "filtering" -- rather, it just adds
+    two new attributes to any "filtered" LogRecord objects:
+
+    * elasticapm_transaction_id
+    * elasticapm_transaction_trace_parent_id
+
+    These attributes can then be incorporated into your handlers and formatters,
+    so that you can tie log messages to transactions in elasticsearch.
+    """
+
+    def filter(self, record):
+        """
+        Add elasticapm attributes to `record`.
+        """
+        transaction = execution_context.get_transaction()
+        record.elasticapm_transaction_id = transaction.id
+        record.elasticapm_transaction_trace_parent_id = (
+            transaction.trace_parent.trace_id if transaction.trace_parent else None
+        )
+        return True
+
+
+def structlog_processor(_, __, event_dict):
+    """
+    Add two new entries to the event_dict for any processed events:
+
+    * elasticapm_transaction_id
+    * elasticapm_transaction_trace_parent_id
+
+    :param _:
+        Unused (logger instance in structlog)
+    :param __:
+        Unused (wrapped method_name)
+    :param event_dict:
+        Event dictionary for the event we're processing
+    :return:
+        `event_dict`, with two new entries.
+    """
+    transaction = execution_context.get_transaction()
+    event_dict["elasticapm_transaction_id"] = transaction.id
```

Comment:
here we should probably use the names suggested by @axw, although I'm not sure if the right thing happens if we use dotted names (as in, does the value end up in the correct field in ES if we configure structlog to output JSON and pipe it into ES via filebeat or somesuch).

Reply:
Still haven't tested, but as far as I can tell [dots in field names are turned into properties in ES](https://www.elastic.co/guide/en/elasticsearch/reference/2.4/dots-in-names.html). So this should be good.